neon_utils/signal_utils.py

```diff
@@ -0,0 +1,167 @@
+# NEON AI (TM) SOFTWARE, Software Development Kit & Application Framework
+# All trademark and other rights reserved by their respective owners
+# Copyright 2008-2022 Neongecko.com Inc.
+# Contributors: Daniel McKnight, Guy Daniels, Elon Gasper, Richard Leeds,
+# Regina Bloomstine, Casimiro Ferreira, Andrii Pernatii, Kirill Hrymailo
+# BSD-3 License
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+from time import time, sleep
+
+from mycroft_bus_client import MessageBusClient, Message
+import ovos_utils.signal
+
+from neon_utils.logger import LOG
+
+BUS = MessageBusClient()
```

Comment:
can we make this a private var and move this to some helper func?

```python
def init_signal_bus(bus):
    global _BUS
    _BUS = bus
```

maybe set it to FakeBus by default before this is called?

Reply:
`Configuration`class does the same, this init method would be called by individual services on launch (skills/audio/speech/enclosure)

Reply:
Would this be called on import? Main concern is compatibility of the signal methods where we don't know if the caller has a bus instance (yet)

Reply:
I could remove the call to `init_signal_handlers` on import and add a `bus` param there.. I suppose this is a new module, so there's not really any backwards-compat to consider

Reply:
the intention is to allow passing a bus connection and share it with parent service, so you could make the proposed  `init_signal_bus` have bus as optional arg. and create the connection there when `bus is None` 

when bus is needed first time if it wasnt set then call `init_signal_bus`, so bus is lazy loaded but still ensured to be there

its better to avoid it being done on import which can cause issues in some scenarios